src/Compilers/VisualBasic/Portable/Syntax/SyntaxFactory.vb

```diff
@@ -532,5 +534,114 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return SyntaxEquivalence.AreEquivalent(oldList.Node, newList.Node, ignoreChildNode, topLevel:=False)
         End Function
 
+        ''' <summary>
+        ''' Determines if a submission contains an LINQ query not followed by an empty new line.
+        '''
+        ''' Examples:
+        ''' 1. <c>Dim x = 1</c> returns false since the statement is not a LINQ query
+        ''' 2. <c>
+        ''' Dim x = FROM {1, 2, 3}
+        '''
+        ''' </c> return false since the statement is followed by an empty new line.
+        ''' 3. <c>Dim x = FROM {1, 2, 3}</c> returns true since the LINQ statement is not followed by an empty new line.
+        ''' </summary>
+        ''' <param name="token">Last expected token of the last statement in a submission.</param>
+        ''' <param name="statementNode">Top level statement which token is part of.</param>
+        ''' <param name="endOfFileToken">Token that marks the end of submission.</param>
+        Private Shared Function IsPartOfLinqQueryNotFollowedByNewLine(token As SyntaxToken, statementNode As SyntaxNode, endOfFileToken As SyntaxToken) As Boolean
+            ' Checking if the submission ends with a new line.
+            For Each leadingTrivia In endOfFileToken.LeadingTrivia
+                If leadingTrivia.IsKind(SyntaxKind.EndOfLineTrivia) Then
+                    Return False
+                End If
+            Next
+
+            ' Checking if the last token is part of a LINQ query.
+            Dim node = token.Parent
+            Do
+                If node.IsKind(SyntaxKind.QueryExpression) Then
+                    Return True
+                End If
+
+                If node Is statementNode Then
+                    Return False
+                End If
+
+                node = node.Parent
+            Loop
+
+            Throw ExceptionUtilities.Unreachable
```

Comment:
The compiler can determine that this is unreachable, can't it?  We generally only throw this when a human can but the compiler can't.